front50-migrations/src/test/groovy/com/netflix/spinnaker/front50/migrations/MoveSubpathToLocationMigrationSpec.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.migrations
+
+import com.netflix.spinnaker.front50.model.pipeline.Pipeline
+import com.netflix.spinnaker.front50.model.pipeline.PipelineDAO
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class MoveSubpathToLocationMigrationSpec extends Specification {
+  def pipelineDAO = Mock(PipelineDAO)
+
+  @Subject
+  def migration = new MoveSubpathToLocationMigration(pipelineDAO)
+
+  @Unroll
+  def "should not set subpath in location"() {
+    given:
+    def pipeline = new Pipeline([application: "test", expectedArtifacts: [[matchArtifact: original]]])
+    pipeline.id = "pipeline-1"
+
+    when:
+    migration.run()
+
+    then:
+    _ * pipelineDAO.all() >> { return [pipeline] }
+    0 * pipelineDAO.update("pipeline-1", _)
+
+    where:
+    original  || _
```

Comment:
This syntax is a little unusual to me (are you taking this approach because you need at least two columns in the table?) If you modeled it after other tests in this project are structured, that's fine.

Otherwise, you can structure it like this, I think:
```
original << [
   [:],
   [type: "bitbucket/file", reference: "test-ref"],
   ...
]
```